spec/jobs/sufia_resque_queue_spec.rb

```diff
@@ -0,0 +1,49 @@
+require 'spec_helper'
+
+describe Sufia::Resque::Queue do
+  let(:subject) { Sufia::Resque::Queue.new "test_queue" }
+  let(:job) { double('job') }
+
+  context "with no retries" do
+    it "queues the job" do
+      expect(::Resque).to receive(:enqueue_to).once.and_return(true)
+      subject.push(job)
+    end
+  end
+
+  context 'when one run times out' do
+    before do
+      call_count = 0
+      allow(::Resque).to receive(:enqueue_to) do
+        call_count += 1
```

Comment:
@mjgiarlo I really liked my syntax much better than this here.  It was really easy (in my mind) to see what was going to happen).  This is very convoluted. That said I can live with this syntax.